llvm/lib/Target/ARM/ARMSubtarget.h

```diff
@@ -150,6 +150,23 @@ class ARMSubtarget : public ARMGenSubtargetInfo {
     SingleIssuePlusExtras,
   };
 
+  /// How the pushing and popping of callee saved registers to and from the
+  /// stack should be split.
+  enum PushPopSplitVariation {
+    /// r4-r11+lr (+r12 if necessary) can be pushed in a single instruction.
+    NoSplit,
+    /// The registers need to be split into a push of r4-r7+lr and another
+    /// containing r8-r11 (+r12 if necessary).
```

Comment:
Maybe worth noting the interaction between R7Split and the Thumb1 AAPCS split push (HasFrameRecordArea).